### ISSUE TITLE
Zeros are truncated from strings.

### ISSUE THREAD

**Issue by dellis23** [ansible/ansible#10281]
##### Issue Type:

Bug Report
##### Ansible Version:

`ansible 1.8.2`
##### Environment:

Mac OSX 10.9 running against Ubuntu 14.04.
##### Summary:

When defining a string, trailing `0`s get removed.
##### Steps To Reproduce:

In my inventory:

```
server apples="1.20"
```

In my playbook:

```
debug msg="{{ apples }}"
```

Output:

```
TASK: [playbook | debug msg="{{ apples }}"] ************************** 
ok: [server] => {
    "msg": "1.2"
}
```
##### Expected Results:

1.20
##### Actual Results:

1.2

**Comment by sivel** [comment]
I cannot duplicate this in either 1.8.2 or current devel.  I always get:

```
ok: [server] => {
    "msg": "20"
}
```

**Comment by dellis23** [comment]
Sorry, I've updated the example.  A co-worker pointed out that this is actually a problem with it (and I don't know what _it_ is) interpreting the value as a float and truncating it accordingly.  I'm not sure where along the chain of ansible -> yaml -> jinja this bug lies.  Here was my solution though:

```
apples="{{ '1.20' }}"
```

**Comment by sivel** [comment]
Ok, that I can duplicate. It looks like the use of `ast.literal_eval` in `ansible.inventory.ini.InventoryParser._parse_value` doesn't treat it correctly.

**Comment by sivel** [comment]
Looks like ast was introduced in https://github.com/ansible/ansible/pull/4268

**Comment by abadger** [comment]
Did a quick test and it does appear that this has been fixed by #10313.  I'm going to close this.  Feel free to open a new bug report if the error continues.

# Closing This Ticket

Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
